offset_request.go

```diff
@@ -12,7 +12,7 @@ func (r *offsetRequestBlock) encode(pe packetEncoder) error {
 }
 
 type OffsetRequest struct {
-	blocks map[string]map[int32]*offsetRequestBlock
+	blocks map[string]map[int32][]*offsetRequestBlock
```

Comment:
Since this is apparently legit (who knew) we should probably update all the other request types as well - perhaps that deserves a separate PR before this one.